File: Godeps/_workspace/src/github.com/yhat/ops2/src/mps/integration/python/svc/svc.py
import os

import numpy as np
import pandas as pd
from sklearn.svm import SVC
from sklearn.datasets import load_iris

from yhat import Yhat, YhatModel, preprocess, df_to_json


iris = load_iris()

X = pd.DataFrame(iris.data, columns=iris.feature_names)
y = pd.DataFrame(iris.target, columns=["flower_types"])

clf = SVC()
clf.fit(X, y["flower_types"])


class MySVC(YhatModel):
    @preprocess(in_type=pd.DataFrame, out_type=pd.DataFrame)
    def execute(self, data):
        prediction = clf.predict(pd.DataFrame(data))
        species = ['setosa', 'versicolor', 'virginica']
        result = [species[i] for i in prediction]
        return result

username = os.environ["USERNAME"]
apikey = os.environ["APIKEY"]
endpoint = os.environ["OPS_ENDPOINT"]

print "%s:%s:%s" % (username, apikey, endpoint,)

yh = Yhat(
    username,
    apikey,
    endpoint
)
yh.deploy("SupportVectorClassifier", MySVC, globals(), sure=True)

File: Godeps/_workspace/src/github.com/yhat/ops2/src/mps/integration/r/svc/svc.r
library(datasets)
library(yhatr)

#import the svm package
library(e1071)

fit <- svm(Species ~ Petal.Width + Sepal.Length + Sepal.Width + Petal.Length, data=iris)

model.require <- function() {
  library(e1071)
}

model.transform <- function(data){
  data
}

model.predict <- function(data) {
  pred <- predict(fit, newdata=data.frame(data))
  result <- data.frame(pred)
  result
}

username <- Sys.getenv("USERNAME")
print(username)
apikey <- Sys.getenv("APIKEY")
print(apikey)
endpoint <- Sys.getenv("OPS_ENDPOINT")
print(endpoint)

yhat.config  <- c(
    username=username,
    apikey=apikey,
    env=endpoint
)

yhat.deploy("SupportVectorClassifierR")
